core/domain/feature_flag_services_test.py

```diff
@@ -0,0 +1,542 @@
+                )
+
+    def test_feature_flag_enabled_for_users_in_5_perc_when_increased_to_10_perc(
+        self) -> None:
+        user_ids_list = []
+        user_id_to_feature_flag_status_for_5_perc = {}
+        count_feature_flag_enabled_for_5_perc = 0
+        user_id_to_feature_flag_status_for_10_perc = {}
+        count_feature_flag_enabled_for_10_perc = 0
+        for count in range(1, 1001):
+            user_ids_list.append('userid' + str(count))
+
+        with self.swap_all_feature_names_set:
+            feature_services.update_feature_flag(
+                self.dev_feature_flag.name, False, 5, [])
+            for user_id in user_ids_list:
+                feature_flag_status = feature_services.is_feature_flag_enabled(
+                    user_id,
+                    self.dev_feature_flag.name
+                )
+                if feature_flag_status:
+                    count_feature_flag_enabled_for_5_perc += 1
+                    user_id_to_feature_flag_status_for_5_perc[
+                        user_id] = feature_flag_status
+            self.assertTrue(
+                count_feature_flag_enabled_for_5_perc in list(range(40, 61)))
+
+            feature_services.update_feature_flag(
+                self.dev_feature_flag.name, False, 10, [])
+            for user_id in user_ids_list:
+                feature_flag_status = feature_services.is_feature_flag_enabled(
+                    user_id,
+                    self.dev_feature_flag.name
+                )
+                if feature_flag_status:
+                    count_feature_flag_enabled_for_10_perc += 1
+                    user_id_to_feature_flag_status_for_10_perc[
+                        user_id] = feature_flag_status
+            self.assertTrue(
+                count_feature_flag_enabled_for_10_perc in list(range(90, 111)))
+
+            for user_id in user_id_to_feature_flag_status_for_5_perc:
+                self.assertIn(
+                    user_id, user_id_to_feature_flag_status_for_10_perc)
+
+    def test_feature_flag_flag_not_enabled_for_all_users(self) -> None:
+        user_1_id, user_2_id, user_3_id = (
+            self._signup_multiple_users_and_return_ids())
+        user_ids = [user_1_id, user_2_id, user_3_id, self.owner_id]
+        feature_status_for_users = []
+        with self.swap_all_feature_names_set:
+            feature_services.update_feature_flag(
+                self.dev_feature_flag.name, False, 0, [])
+            for user_id in user_ids:
+                feature_status_for_users.append(
+                    feature_services.is_feature_flag_enabled(
+                        user_id, self.dev_feature_flag.name))
+
+        total_count_of_users_having_feature_enabled = 0
+        for feature_status in feature_status_for_users:
+            if feature_status is True:
+                total_count_of_users_having_feature_enabled += 1
+
+        self.assertEqual(total_count_of_users_having_feature_enabled, 0)
```

Comment:
Perhaps just assert instead, in the for loop above, that feature_status is False for everyone?

Reply:
Updated